### PR TITLE
soundwire: initial work

### PR DESCRIPTION
**Pull request by slawblauciak**
Provides basic structure work for future SoundWire implementation.

Signed-off-by: Slawomir Blauciak <slawomir.blauciak@linux.intel.com>

### CODE REVIEW

File: src/drivers/intel/cavs/soundwire.c [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+#define trace_soundwire(__e, ...) \
+	trace_event(TRACE_CLASS_SSP, __e, ##__VA_ARGS__)
+#define trace_soundwire_error(__e, ...)	\
+	trace_error(TRACE_CLASS_SSP, __e, ##__VA_ARGS__)
+#define tracev_soundwire(__e, ...) \
+	tracev_event(TRACE_CLASS_SSP, __e, ##__VA_ARGS__)
+
+static int soundwire_trigger(struct dai *dai)
+{
+	return 0;
```
Comment (jajanusz) [inline src/drivers/intel/cavs/soundwire.c]: Shouldn't we return ENOSYS or EINVAL here when function is not implemented? I know we already return 0 in other interfaces, but maybe it's better to let user know that he called something that is not there yet.
@mmaka1 @lgirdwood 
What do you think?

Comment (plbossart) [reply src/drivers/intel/cavs/soundwire.c]: It's fine by me for starters, but having a trace would be useful to check which code paths are taken.

**Comment by xiulipan** [comment]
SOFCI TEST

**Comment by sofci** [comment]
Can one of the admins verify this patch?

File: src/drivers/intel/cavs/soundwire.c (hunk 1 above)
Comment (jajanusz) [reply src/drivers/intel/cavs/soundwire.c]: @slawblauciak Please add traces at least and then I'll approve

**Pull request merged**